lib/rules/media-feature-range-notation/index.js

```diff
@@ -33,57 +40,101 @@ const rule = (primary) => {
 		}
 
 		root.walkAtRules(/^media$/i, (atRule) => {
-			mediaParser(atRule.params).walk(/^media-feature$/i, ({ parent, value }) => {
-				if (!isStandardSyntaxMediaFeatureName(value)) return;
-
-				if (!isRangeContextMediaFeature(value) && isInBooleanContext(parent)) return;
-
-				if (!isRangeContextMediaFeature(value) && !isRangeTypeMediaFeature(value)) return;
-
-				if (primary === 'prefix' && isPrefixedRangeMediaFeature(value)) return;
+			const mediaQueryList = parseMediaQuery(atRule, result);
+			let hasFixes = false;
+
+			mediaQueryList.forEach((mediaQuery) => {
+				mediaQuery.walk((entry) => {
+					const node = entry.node;
+
+					// Only look at plain and range notation media features
+					if (!isMediaFeatureRange(node) && !isMediaFeaturePlain(node)) return;
+
+					// Expected plain notation and received plain notation
+					if (primary === 'prefix' && isMediaFeaturePlain(node)) return;
+
+					// Expected range notation and received range notation
+					if (primary === 'context' && isMediaFeatureRange(node)) return;
+
+					const featureName = node.getName();
+					const unprefixedMediaFeature = featureName.replace(/^(?:min|max)-/i, '');
+
+					if (!rangeTypeMediaFeatureNames.has(unprefixedMediaFeature)) return;
+
+					if (context.fix && primary === 'context' && isMediaFeaturePlain(node)) {
+						hasFixes = true;
+
+						/** @type {import('@csstools/css-tokenizer').TokenDelim} */
+						let operator = [TokenType.Delim, '<', -1, -1, { value: '<' }];
+
+						if (/^min-/i.test(featureName)) {
+							operator = [TokenType.Delim, '>', -1, -1, { value: '>' }];
+						}
+
+						const parent = entry.parent;
+
+						if (!isMediaFeature(parent)) return;
+
+						parent.feature = new MediaFeatureRangeNameValue(
+							new MediaFeatureName(
+								new TokenNode([
+									TokenType.Ident,
+									unprefixedMediaFeature,
+									-1,
+									-1,
+									{ value: unprefixedMediaFeature },
+								]),
+								node.name.before,
+								node.name.after.length > 0
+									? node.name.after
+									: [[TokenType.Whitespace, ' ', -1, -1, undefined]],
+							),
+							[operator, [TokenType.Delim, '=', -1, -1, { value: '=' }]],
+							node.value,
+						);
+
+						return;
+					}
+
+					const tokens = node.tokens();
+					const firstToken = tokens[0];
+					const lastToken = tokens[tokens.length - 1];
+
+					let startIndex = 0;
+					let endIndex = atRule.params.length;
+
+					if (firstToken && lastToken) {
+						startIndex = firstToken[2];
+						endIndex = lastToken[3];
+					}
+
+					const atRuleIndex = atRuleParamIndex(atRule);
+
+					report({
+						message: messages.expected,
+						messageArgs: [primary],
+						node: atRule,
+						index: atRuleIndex + startIndex - 1,
+						endIndex: atRuleIndex + endIndex + 1 + 1,
+						result,
+						ruleName,
+					});
+				});
+			});
 
-				if (primary === 'context' && isRangeContextMediaFeature(value)) return;
+			if (context.fix && hasFixes) {
```

Comment:
[trivial] `context.fix` here seems duplicate and needless, but it's trivial so you can also leave it as-is.

```suggestion
			if (hasFixes) {
```